src/perception/DetectedObject.cpp

```diff
@@ -7,12 +7,12 @@ namespace perception {
 
 //==============================================================================
 DetectedObject::DetectedObject(
-    const std::string& uid,
-    const std::string& assetKey,
+    const std::string& objectName,
+    const int& objectId,
     const std::string& detectionFrameID,
     const std::string& yamlStr)
-  : mUid(std::move(uid))
-  , mAssetKey(std::move(assetKey))
+  : mObjectName(objectName)
```

Comment:
You still need a default AssetKey here! Otherwise, it will become undefined / 0 at Line 22.
```suggestion
  : mObjectName(objectName)
  , mAssetKey(objectName)
```

Reply:
Please see the updated constructor and my comment above.